zerver/tests/test_message_fetch.py

```diff
@@ -2787,6 +2787,129 @@ def messages_matches_ids(messages: List[Dict[str, Any]], message_ids: List[int])
         self.assertEqual(data["found_newest"], True)
         self.assertEqual(data["history_limited"], False)
 
+    def test_first_after_date_anchor_target_date(self) -> None:
+        """
+        Test getting first message after certain datetime
+        """
+        me = self.example_user("hamlet")
+        aaron = self.example_user("aaron")
+        self.subscribe(me, "Scotland")
+        self.subscribe(aaron, "Scotland")
+        self.login("hamlet")
+
+        target_date = timezone_now()
+
+        topic = "test first message after time"
+        message_ids = []
+        message_ids.append(self.send_stream_message(me, "Scotland", topic_name=topic))
+        for i in range(3):
+            message_ids.append(self.send_stream_message(aaron, "Scotland", topic_name=topic))
+
+        # Edit timestamps to ensure they aren't same second
+        time_delta = 10
+        for msg_id in message_ids:
+            message = Message.objects.get(id=msg_id)
+            message.date_sent = message.date_sent + datetime.timedelta(seconds=time_delta)
+            message.save()
+            time_delta += 10
+
+        narrow = [
+            dict(operator="stream", operand="Scotland"),
+            dict(operator="topic", operand=topic),
+        ]
+        params = dict(
+            narrow=orjson.dumps(narrow).decode(),
+            anchor="first_after_date",
+            target_date=target_date.isoformat(),
+            num_before=0,
+            num_after=0,
+        )
+        result = self.client_get("/json/messages", params)
+        self.assert_json_success(result)
+        result = result.json()
+        self.assert_length(result["messages"], 1)
+        self.assertEqual(result["messages"][0]["id"], message_ids[0])
+        self.assertEqual(result["anchor"], message_ids[0])
+
+    def test_first_after_date_anchor_different_narrow_first_msg(self) -> None:
```

Comment:
duplicates code from test_first_after_date_anchor_target_date ?  refactor into a common setup function ?

Reply:
I think little duplication is OK in tests. I feel like we'd be complicating it trying to refactor.